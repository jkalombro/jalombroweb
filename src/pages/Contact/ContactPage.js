import React from "react";
import "./ContactPage.scss";
import { useRef } from "react";
import emailjs from "@emailjs/browser";
import "react-toastify/dist/ReactToastify.min.css";
import { ToastContainer, toast } from "react-toastify";

import MyAddressMap from "../../components/MyAddressMap";

const ContactPage = () => {
    const form = useRef();

    const sendEmail = async e => {
        e.preventDefault();

        var res = await toast.promise(emailjs.sendForm("service_srpy7bf", "template_mx97uen", form.current, "Uw39eNBn1znHrYBfB"), {
            pending: "Sending Messages..",
            success: "Message successfully sent!",
            error: "Failed to send the message, please try again.",
        });

        //resets the form if sending is success
        res && res?.status === 200 && e.target.reset();
    };

    return (
        <section className="contact-page" data-testid="contactpage">
            <section className="form-col">
                <h1>Contact Me</h1>
                <p>If you're interested on me and you want to reach out, don't hesitate to fill-up the email form below and send me a message. Can't wait to discuss with you soon.</p>
                <div className="contact-form">
                    <form ref={form} onSubmit={sendEmail}>
                        <ul>
                            <li className="half">
                                <input placeholder="Name" type="text" name="name" autoComplete="off" required />
                            </li>
                            <li className="half">
                                <input placeholder="Email" type="email" name="email" autoComplete="off" required />
                            </li>
                            <li>
                                <input placeholder="Subject" type="text" name="subject" autoComplete="off" required />
                            </li>
                            <li>
                                <textarea placeholder="Message" name="message" required></textarea>
                            </li>
                            <li>
                                <button type="submit" className="flat-button">
                                    SEND
                                </button>
                            </li>
                        </ul>
                    </form>
                </div>
            </section>

            <section className="map-col">
                <MyAddressMap />
            </section>

            <ToastContainer position="top-center" autoClose={5000} hideProgressBar={false} newestOnTop={true} theme="dark" closeOnClick rtl={false} pauseOnFocusLoss draggable pauseOnHover />
        </section>
    );
};

export default ContactPage;
