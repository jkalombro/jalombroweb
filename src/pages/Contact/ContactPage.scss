@import "../../App.scss";

section.contact-page {
  @include common-page-attr;
  min-height: 90vh;
  color: #e9e9ea;
  overflow: hidden;

  .form-col,
  .map-col {
    flex: 1;
    width: 50%;
    padding: 50px;
    display: flex;
    flex-direction: column;
    align-content: center;
    justify-content: center;
  }

  .form-col {
    p {
      font-size: 20px;
      animation: fadeIn 1s;
      opacity: 0.8;
    }

    ul {
      padding: 0;
      margin: 0;

      li {
        padding: 0;
        margin: 0;
        list-style: none;
        margin-bottom: 10px;
        opacity: 0;
        overflow: hidden;
        display: block;
        clear: both;
        position: relative;
        animation: fadeIn 1s;
        animation-fill-mode: forwards;
      }

      li.half {
        width: 49%;
        margin-left: 2%;
        float: left;
        clear: none;

        &:first-child {
          margin-left: 0;
        }
      }
    }

    input[type="text"],
    input[type="email"] {
      font-family: "Lato", sans-serif !important;
      width: 100%;
      border: 0;
      background: #585f6e;
      height: 50px;
      font-size: 16px;
      color: #fff;
      padding: 0 20px;

      &::placeholder {
        color: #fff;
        opacity: 0.7;
      }
    }

    textarea {
      font-family: "Lato", sans-serif !important;
      width: 100%;
      border: 0;
      background: #585f6e;
      height: 50px;
      font-size: 16px;
      color: #fff;
      padding: 20px;
      min-height: 150px;

      &::placeholder {
        color: #fff;
        opacity: 0.7;
      }
    }

    .flat-button {
      color: #60c3eb;
      font-size: 15px;
      letter-spacing: 3px;
      text-decoration: none;
      padding: 15px 25px;
      border: 1px solid #60c3eb;
      border-radius: 4px;
      background: 0 0;
      text-transform: uppercase;
      float: right;
      text-align: center;
      margin-right: 10px;
      cursor: pointer;

      &:hover,
      &:focus,
      &:active {
        background: #20232a;
      }
    }
  }

  .map-col {
    flex: 1;
    width: 50%;
    position: relative;
  }
}

@include mobile {
  section.contact-page {
    flex-direction: column;

    .form-col,
    .map-col {
      width: 90vw;
      padding: 20px;
    }

    .form-col {
      p {
        font-size: 16px;
      }
    }
  }
}
