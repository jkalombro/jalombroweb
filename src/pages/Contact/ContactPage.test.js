import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, act } from "@testing-library/react";
import render from "../../utils/jest-context-helper";
import ContactPage from "./ContactPage";

describe("ContactPage Component", () => {
  it("renders without crashing", () => {
    const { getByTestId } = render(<ContactPage />);
    const element = getByTestId("contactpage");
    expect(element).toBeTruthy();
  });

  it("matches with snapshot", () => {
    const { getByTestId } = render(<ContactPage />);
    const element = getByTestId("contactpage");
    expect(element).toMatchSnapshot();
  });

  it("initializes an empty form", () => {
    const { getByPlaceholderText } = render(<ContactPage />);
    const nameinput = getByPlaceholderText(/name/i);
    const emailinput = getByPlaceholderText(/email/i);
    const subjectinput = getByPlaceholderText(/subject/i);
    const messageinput = getByPlaceholderText(/message/i);
    expect(nameinput).toHaveValue("");
    expect(emailinput).toHaveValue("");
    expect(subjectinput).toHaveValue("");
    expect(messageinput).toHaveValue("");
  });

  it("handles Name input correctly", async () => {
    const { getByPlaceholderText } = render(<ContactPage />);
    const input = getByPlaceholderText(/name/i);
    let sample = "Sample Name";
    await fireEvent.change(input, { target: { value: sample } });
    expect(input).toHaveValue(sample);
    expect(input).toBeValid();
    await fireEvent.change(input, { target: { value: "" } });
    expect(input).toHaveValue("");
    expect(input).toBeInvalid();
  });

  it("handles Email input correctly", async () => {
    const { getByPlaceholderText } = render(<ContactPage />);
    const input = getByPlaceholderText(/email/i);
    let sample = "sample email";
    await fireEvent.change(input, { target: { value: sample } });
    expect(input).toHaveValue(sample);
    expect(input).toBeInvalid();
    sample = "correctemail@gmail.com";
    await fireEvent.change(input, { target: { value: sample } });
    expect(input).toHaveValue(sample);
    expect(input).toBeValid();
  });

  it("handles Subject input correctly", async () => {
    const { getByPlaceholderText } = render(<ContactPage />);
    const input = getByPlaceholderText(/subject/i);
    let sample = "Sample Subject";
    await fireEvent.change(input, { target: { value: sample } });
    expect(input).toHaveValue(sample);
    expect(input).toBeValid();
    await fireEvent.change(input, { target: { value: "" } });
    expect(input).toHaveValue("");
    expect(input).toBeInvalid();
  });

  it("handles Message input correctly", async () => {
    const { getByPlaceholderText } = render(<ContactPage />);
    const input = getByPlaceholderText(/message/i);
    let sample = "Sample Message hehe";
    await fireEvent.change(input, { target: { value: sample } });
    expect(input).toHaveValue(sample);
    expect(input).toBeValid();
  });

  it("will clear the form when submit is success", async () => {
    const { getByPlaceholderText, getByRole, findByPlaceholderText } = render(
      <ContactPage />
    );
    const nameinput = getByPlaceholderText(/name/i);
    const emailinput = getByPlaceholderText(/email/i);
    const subjectinput = getByPlaceholderText(/subject/i);
    const messageinput = getByPlaceholderText(/message/i);
    const button = getByRole("button", {
      name: /send/i,
    });

    await act(async () => {
      await fireEvent.change(nameinput, { target: { value: "Test Name" } });
      await fireEvent.change(emailinput, {
        target: { value: "test@gmail.com" },
      });
      await fireEvent.change(subjectinput, {
        target: { value: "Test Subject" },
      });
      await fireEvent.change(messageinput, {
        target: { value: "Test Message" },
      });
      await fireEvent.click(button);

      await new Promise((r) => setTimeout(r, 2000));

      const newname = await findByPlaceholderText(/name/i);
      const newemail = await findByPlaceholderText(/email/i);
      const newsubject = await findByPlaceholderText(/subject/i);
      const newmessage = await findByPlaceholderText(/message/i);
      expect(newname).toHaveValue("");
      expect(newemail).toHaveValue("");
      expect(newsubject).toHaveValue("");
      expect(newmessage).toHaveValue("");
    });
  });

//   it("will NOT clear the form when submit is failed", async () => {
//     const { getByPlaceholderText, getByRole, findByPlaceholderText } = render(
//       <ContactPage />
//     );
//     const nameinput = getByPlaceholderText(/name/i);
//     const emailinput = getByPlaceholderText(/email/i);
//     const subjectinput = getByPlaceholderText(/subject/i);
//     const messageinput = getByPlaceholderText(/message/i);
//     const button = getByRole("button", {
//       name: /send/i,
//     });

//     const samplename = "Test Name";
//     const sampleemail = "";
//     const samplesubject = "Test Subject";
//     const samplemsg = "Test Message";

//     await act(async () => {
//       await fireEvent.change(nameinput, { target: { value: samplename } });
//       await fireEvent.change(emailinput, { target: { value: sampleemail } });
//       await fireEvent.change(subjectinput, { target: { value: samplesubject } });
//       await fireEvent.change(messageinput, { target: { value: samplemsg } });
//       await fireEvent.click(button);

//       await new Promise((r) => setTimeout(r, 2000));

//       const newname = await findByPlaceholderText(/name/i);
//       const newemail = await findByPlaceholderText(/email/i);
//       const newsubject = await findByPlaceholderText(/subject/i);
//       const newmessage = await findByPlaceholderText(/message/i);
//       expect(newname).toHaveValue(samplename);
//       expect(newemail).toHaveValue(sampleemail);
//       expect(newemail).toBeInvalid();
//       expect(newsubject).toHaveValue(samplesubject);
//       expect(newmessage).toHaveValue(samplemsg);
//     });
//   });
});
