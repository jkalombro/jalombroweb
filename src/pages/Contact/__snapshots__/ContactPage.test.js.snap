// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage Component matches with snapshot 1`] = `
<section
  class="contact-page"
  data-testid="contactpage"
>
  <section
    class="form-col"
  >
    <h1>
      Contact Me
    </h1>
    <p>
      If you're interested on me and you want to reach out, don't hesitate to fill-up the email form below and send me a message. Can't wait to discuss with you soon.
    </p>
    <div
      class="contact-form"
    >
      <form>
        <ul>
          <li
            class="half"
          >
            <input
              autocomplete="off"
              name="name"
              placeholder="Name"
              required=""
              type="text"
            />
          </li>
          <li
            class="half"
          >
            <input
              autocomplete="off"
              name="email"
              placeholder="Email"
              required=""
              type="email"
            />
          </li>
          <li>
            <input
              autocomplete="off"
              name="subject"
              placeholder="Subject"
              required=""
              type="text"
            />
          </li>
          <li>
            <textarea
              name="message"
              placeholder="Message"
              required=""
            />
          </li>
          <li>
            <button
              class="flat-button"
              type="submit"
            >
              SEND
            </button>
          </li>
        </ul>
      </form>
    </div>
  </section>
  <section
    class="map-col"
  >
    <div
      data-testid="addressMap"
    >
      <div
        class="info-map"
      >
        Villa Azalea, Cotcot
        <br />
        Liloan, Cebu
        <br />
        Philippines, 6002 
        <br />
        <br />
        <span>
          jerichalombro@gmail.com
        </span>
      </div>
      <div
        class="map-wrap"
      >
        <div
          class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
          style="position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 1;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://a.tile.openstreetmap.org/9/432/240.png"
                    style="width: 256px; height: 256px; left: -72px; top: -240px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-shadow-pane"
            >
              <img
                alt=""
                class="leaflet-marker-shadow leaflet-zoom-hide"
                src="marker-shadow.png"
                style="left: 0px; top: 0px;"
              />
            </div>
            <div
              class="leaflet-pane leaflet-marker-pane"
            >
              <img
                alt="Marker"
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                role="button"
                src="marker-icon.png"
                style="left: 0px; top: 0px; z-index: 0;"
                tabindex="0"
              />
            </div>
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            />
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JavaScript library for interactive maps"
                >
                  Leaflet
                </a>
                 
                <span
                  aria-hidden="true"
                >
                  |
                </span>
                 Â© 
                <a
                  href="https://www.openstreetmap.org/copyright"
                >
                  OpenStreetMap
                </a>
                 contributors
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="Toastify"
  />
</section>
`;
